src/Microsoft.Framework.Runtime/ProjectResolver.cs

```diff
@@ -37,14 +37,21 @@ public bool TryResolveProject(string name, out Project project)
         {
             project = null;
 
-            ProjectInformation projectInfo;
-            if (_projects.TryGetValue(name, out projectInfo))
+            if (!_projects.Contains(name))
             {
-                project = projectInfo.Project;
-                return project != null;
+                return false;
             }
 
-            return false;
+            var candidates = _projects[name].Where(p => File.Exists(Path.Combine(p.FullPath, Project.ProjectFileName)));
```

Comment:
This code paths runs every time somebody asks for a project. Checking the existence is why ProjectInformation.Project is lazy and only ever runs once.